src/Extension/AccountReset/SecurityExtension.php

```diff
@@ -0,0 +1,160 @@
+
+/**
+ * Extends the Security controller to support Account Resets. This extension can
+ * itself be extended to add procedures to the reset action (such as removing
+ * additional authentication factors, sending alerts, etc.)
+ *
+ * @package SilverStripe\MFA\Extension
+ * @property Security owner
+ */
+class SecurityExtension extends Extension
+{
+    use JSONResponse, BaseHandlerTrait, Extensible;
+
+    private static $url_handlers = [
+        'GET reset-account' => 'resetaccount',
+        'ResetAccountForm' => 'ResetAccountForm',
+    ];
+
+    private static $allowed_actions = [
+        'resetaccount',
+        'ResetAccountForm',
+    ];
+
+    public function resetaccount(HTTPRequest $request): HTTPResponse
+    {
+        if (Security::getCurrentUser()) {
+            return $this->jsonResponse(['error' => 'Already authenticated'], 400);
+        }
+
+        ['m' => $memberID, 't' => $token] = $request->getVars();
+
+        /** @var Member|MemberExtension $member */
+        $member = Member::get()->byID($memberID);
+
+        if (is_null($member) || $member->verifyAccountResetToken($token) === false) {
+            return $this->jsonResponse(['error' => 'Invalid member or token'], 400);
+        }
+
+        $request->getSession()->set('MemberID', $member->ID);
+
+        return $this->owner->getResponse()->setBody($this->owner->renderWith(
+            'Security',
+            [
+                'Title' => _t(__CLASS__ . '.ACCOUNT_RESET_TITLE', 'Reset Account'),
+                'Message' => _t(__CLASS__ . '.ACCOUNT_RESET_DESCRIPTION', 'Your password will be changed, and any registered MFA methods will be removed.'),
+                'Form' => $this->ResetAccountForm(),
+            ]
+        ));
+    }
+
+    public function ResetAccountForm(): Form
+    {
+        $fields = FieldList::create([
+            PasswordField::create('NewPassword1', _t('SilverStripe\\Security\\Member.NEWPASSWORD', 'New Password')),
+            PasswordField::create('NewPassword2', _t('SilverStripe\\Security\\Member.CONFIRMNEWPASSWORD', 'Confirm New Password'))
+        ]);
+
+        $actions = FieldList::create([
+            FormAction::create('doResetAccount', 'Reset Account'),
+        ]);
+
+        $validation = RequiredFields::create(['NewPassword1', 'NewPassword2']);
+
+        $form = Form::create($this->owner, 'ResetAccountForm', $fields, $actions, $validation);
+
+        $this->owner->extend('updateResetAccountForm', $form);
+
+        return $form;
+    }
+
+    /**
+     * Resets the user's password, and triggers other account reset procedures
+     *
+     * @param array $data
+     * @param Form $form
+     * @return HTTPResponse
+     */
+    public function doResetAccount(array $data, Form $form): HTTPResponse
+    {
+        $memberID = $this->owner->getRequest()->getSession()->get('MemberID');
+
+        // If the ID isn't in the session, politely assume the session has expired
+        if (!$memberID) {
```

Comment:
Do we need sudo mode and CSRF protection here?

Reply:
See above comment about sudo mode. CSRF protection is enabled by default on Forms (I've tested this as working on Account Reset).